### DISCUSSION TITLE
File Support

### DISCUSSION THREAD

**Discussion by kleozzy** [danielmiessler/Fabric#59] [Q&A]
Hello and excellent work !

I was wondering if it is somehow possible to use fabric on files just like i would use OpenAI's Data analytics upload functionality and analyze , extract and manipulate data from files?

Thanks in advance.

**Comment by danielmiessler** [comment]
Interesting!

Can you give us some examples?

**Comment by kleozzy** [comment]
Sure !

The idea is to add the ability to do things like :

echo "{folder_path}" | fabric --some_folder_flag_name --pattern "find what are the total cost for company X in the month of Jan" .

OR

echo "{file_path}" | fabric --readfile_flag --patern "do something" - example : pass a video file , ask fabric to create subs .

**Reply by nic0711** [reply]
You may have to vectorize the documents - but this is actually a matter for the LLM.
However, it is very exciting and would take the whole thing to a new level

**Reply by phoenix21st** [reply]
I was wondering if kleozzy suggestion was incorporated into fabric? Thank you.

**Reply by ksylvan** [reply]
Depending on the provider, you can attach files to your prompt using the `--attach` flag and ask questions about it, yes.

**Reply by phoenix21st** [reply]
Thank you for the reply. I am very new to fabric . . . I could not figure out using --attachment flag. However, after converting .docx to .txt, I made work with _**cat <filename> | fabric -p extract_wisdom**_ as suggested above. It did not work with PDF nor .docx.

**Reply by ksylvan** [reply]
> Thank you for the reply. I am very new to fabric . . . I could not figure out using --attachment flag. However, after converting .docx to .txt, I made work with _**cat <filename> | fabric -p extract_wisdom**_ as suggested above. It did not work with PDF nor .docx.

Which models are you using?

It works with some models and not others, depending. 

Feel free to create an issue in the repo if you know a provider has the capability but Fabric isn't working correctly.
